src/components/GoodsList/GoodsList.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class GoodsList extends React.Component {
+  state = {
+    goods: [...this.props.listOfGoods],
+    defaultSelectValue: 1,
+  }
+
+  array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+  reverceGoods = () => {
+    this.setState(prev => ({
+      goods: [...prev.goods].reverse(),
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(prev => ({
+      goods: [...prev.goods].sort(),
+    }));
+  }
+
+  reset = () => {
+    this.setState(prev => ({
+      goods: [...this.props.listOfGoods],
+      defaultSelectValue: 1,
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(prev => ({
+      goods: [...prev.goods].sort((a, b) => (b.length - a.length)),
+    }));
+  }
+
+  lenghFilter = ({ target }) => {
+    this.setState(() => ({
+      goods: [...this.props.listOfGoods].filter(a => a.length >= target.value),
```

Comment:
filter doesn't mutate the initial array so there is no need to copy it, filter does it for you